src/cli/main.c

```diff
@@ -66,18 +82,44 @@ static void memcpyAndAppendNul(char* dst, char* src, int len) {
     dst[len] = '\0';
 }
 
+// getDictuPath sets up the path to be used for saving history
+// amongst other things. The returned string needs to be freed
+// by the caller.
+char *getDictuPath() {
+    char *dictuPath = calloc(PATH_MAX, sizeof(char) * PATH_MAX);
+    char *tmp;
+
+    if ((tmp = getenv("DICTU_PATH")) != NULL ) {
+        strncpy(dictuPath, tmp, strlen(tmp));
+    } else {
+        const char *home = getenv("HOME");
+        strncpy(dictuPath, home, strlen(home));
+        strncat(dictuPath, DICTU_HOME, strlen(DICTU_HOME));
+    }
+
+    return dictuPath;
+}
+
 static void repl(DictuVM *vm) {
     printf(DICTU_STRING_VERSION);
     char *line;
-    linenoiseHistoryLoad("history.txt");
+
+    char *dictuPath = getDictuPath();
+    // create ${HOME}/.dictu. Silently fail if 
+    // already exists.
+    mkdir(dictuPath, 0700);
```

Comment:
I wonder if it's worth checking if it can't create this directory and throw a warning?